python/metatensor_learn/metatensor/learn/nn/_module.py

```diff
@@ -85,8 +85,8 @@ def _apply(self, fn, recurse=True):
         return result
 
 
-# WARNING: used by metatensor.torch.learn, do not update without increasing the version
-# number as for a breaking change.
+# WARNING: this is duplicated in metatensor.torch._module, make sure to change both
+# versions of the function at the same time
```

Comment:
Does it make sense to have an API version string or something to check that it is the same as in the duplicated code? (same comment for the other duplicates)

Reply:
but this could be for a runtime check across versions right? So the user wouldn't have to worry about differing copies?